src/Tools/dotnet-monitor/CollectionRules/ActionOptionsDependencyAnalyzer.cs

```
@@ -0,0 +1,255 @@
+
+                    if (!GetActionResultReference(actionAndResult, actionIndex, out CollectionRuleActionOptions dependencyOptions, out string actionResultName))
+                    {
+                        continue;
+                    }
+
+                    if (propertyDependency == null)
+                    {
+                        propertyDependency = GetOrCreateDependency(actionIndex, property, originalValue);
+                    }
+
+                    var dependency = new ActionDependency(dependencyOptions, actionResultName);
+                    propertyDependency.ActionDependencies.Add(dependency);
+                }
+            }
+        }
+
+        private PropertyDependency GetOrCreateDependency(int actionIndex, PropertyInfo propertyInfo, string originalValue)
+        {
+            if (!_dependencies.TryGetValue(actionIndex, out Dictionary<string, PropertyDependency> properties))
+            {
+                properties = new Dictionary<string, PropertyDependency>();
+                _dependencies.Add(actionIndex, properties);
+            }
+            if (!properties.TryGetValue(propertyInfo.Name, out PropertyDependency dependency))
+            {
+                dependency = new PropertyDependency(propertyInfo, originalValue);
+                properties.Add(propertyInfo.Name, dependency);
+            }
+            return dependency;
+        }
+
+        private bool GetActionResultReference(string actionReference, int actionIndex,
+            out CollectionRuleActionOptions action, out string actionResultName)
+        {
+            action = null;
+            actionResultName = null;
+
+            string[] parts = actionReference.Split(Separator);
+            if (parts.Length != 2)
+            {
+                _ruleContext.Logger.InvalidActionReference(actionReference);
+                return false;
+            }
+
+            string name = parts[0];
```

Check that name is not null or empty and fail early.